app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/SettingsActivity.java

```diff
@@ -301,6 +355,109 @@ public boolean onPreferenceChange(Preference preference, Object newVal) {
         audioPlayer.setDefaultValue(newValues[0]);
     }
 
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        if (requestCode == FILE_REQUEST_CODE && intent != null) {
+            Uri uri = intent.getData();
+            getContentResolver().takePersistableUriPermission(uri, Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
+            PreferenceManager
+                    .getDefaultSharedPreferences(this)
+                    .edit()
+                    .putString(GBPrefs.AUTO_EXPORT_LOCATION, uri.toString())
+                    .apply();
+            String summary = getAutoExportLocationSummary();
+            findPreference(GBPrefs.AUTO_EXPORT_LOCATION).setSummary(summary);
+            boolean autoExportEnabled = GBApplication
+                    .getPrefs().getBoolean(GBPrefs.AUTO_EXPORT_ENABLED, false);
+            int autoExportPeriod = GBApplication
+                    .getPrefs().getInt(GBPrefs.AUTO_EXPORT_INTERVAL, 0);
+            PeriodicExporter.sheduleAlarm(getApplicationContext(), autoExportPeriod, autoExportEnabled);
+        }
+    }
+
+    /*
+    Either returns the file path of the selected document, or the display name
+     */
+    private String getAutoExportLocationSummary() {
+        String autoExportLocation = GBApplication.getPrefs().getString(GBPrefs.AUTO_EXPORT_LOCATION, null);
+        if (autoExportLocation == null) {
+            return "";
+        }
+        Uri uri = Uri.parse(autoExportLocation);
+        try {
+            String filePath = getFilePath(getApplicationContext(), uri);
+            if (filePath != null) {
+                return filePath;
+            }
+        } catch (URISyntaxException e) {
+            return "";
+        }
+        Cursor cursor = getContentResolver().query(
+                uri,
+                new String[] { DocumentsContract.Document.COLUMN_DISPLAY_NAME },
+        null, null, null, null
+        );
+        if (cursor != null && cursor.moveToFirst()) {
+            return cursor.getString(cursor.getColumnIndex(DocumentsContract.Document.COLUMN_DISPLAY_NAME));
+        }
+        return "";
+    }
+
+    /*
+    As seen on stackoverflow https://stackoverflow.com/a/36714242/1207186
+    Try to find the file path of a document uri
+     */
+    private String getFilePath(Context context, Uri uri) throws URISyntaxException {
```

Comment:
We can make that a static method in class `AndroidUtils`.

Reply:
Done! :)